### PR TITLE
Pass query hints down into remote read query proto

### PR DESCRIPTION
**Pull request by henridf**
Completes  #2580

This currently doesn't include tests, but I'd gladly add some if there are suggestions on how/where to do this.

### CODE REVIEW

File: prompb/remote.proto [hunk 1]
```diff
@@ -35,6 +35,7 @@ message Query {
   int64 start_timestamp_ms = 1;
   int64 end_timestamp_ms = 2;
   repeated prometheus.LabelMatcher matchers = 3;
+  prometheus.ReadParams params = 4;
```
Comment (brian-brazil) [inline prompb/remote.proto]: Hints would be clearer than params I think

File: prompb/types.proto [hunk 2]
```diff
@@ -49,3 +49,8 @@ message LabelMatcher {
   string name  = 2;
   string value = 3;
 }
+
+message ReadParams {
+  int64 step = 1;
```
Comment (brian-brazil) [inline prompb/types.proto]: You should mention the unit here

File: prompb/remote.proto (hunk 1 above)
Comment (henridf) [reply prompb/remote.proto]: I modeled this on `SelectParams`, but I agree that Hints is clearer; changed. 
(we should maybe also rename `SelectParams` -> `SelectHints`)

File: prompb/types.proto (hunk 2 above)
Comment (henridf) [reply prompb/types.proto]: done

File: prompb/types.proto [hunk 3]
```diff
@@ -49,3 +49,8 @@ message LabelMatcher {
   string name  = 2;
   string value = 3;
 }
+
+message ReadHints {
+  int64 step_ms = 1; // Query step size in milliseconds.
+  string func = 2;   // String representation of surrounding function or aggregation.
```
Comment (AlekSi) [inline prompb/types.proto]: Could it be any function? Or we should limit it to some small set common for remote storages? Maybe enum will be better?

**Comment by AlekSi** [comment]
I looked into Prometheus code, and it is not clear to me if those hints can be ignored by remote storage. Can they?

File: prompb/types.proto (hunk 3 above)
Comment (brian-brazil) [reply prompb/types.proto]: It's only a hint, and it can be any function.

Comment (henridf) [reply prompb/types.proto]: @AlekSi string vs enum was discussed at #2580, this PR follows that

**Comment by henridf** [comment]
@brian-brazil i addressed your comments, anything further needed to merge this PR?

**Comment by brian-brazil** [comment]
Thanks!

**Comment by jiacai2050** [comment]
@henridf @AlekSi I still have no idea if hints can be ignored by remote storage. Can they?

**Pull request merged**